astropy/units/distribution.py

```diff
@@ -0,0 +1,427 @@
+            reprarr = reprarr[firstparen:]
+            return '{}{} with n_samples={}'.format(self.__class__.__name__,
+                                                    reprarr, self.n_samples)
+            return reprarr
+
+    def __str__(self):
+        distrstr = str(self.distribution)
+        toadd = ' with n_samples={}'.format(self.n_samples)
+        return distrstr + toadd
+
+    def _repr_latex_(self):
+        superlatex = self.distribution._repr_latex_()
+        toadd = r', \; n_{{\rm samp}}={}'.format(self.n_samples)
+        return superlatex[:-1] + toadd + superlatex[-1]
+
+    @property
+    def n_samples(self):
+        """
+        The number of samples of this distribution.  A single int.
+        """
+        return self.dtype['samples'].shape[0]
+
+    @property
+    def pdf_mean(self):
+        """
+        The mean of this distribution.
+        """
+        return self.distribution.mean(axis=-1)
+
+    @property
+    def pdf_std(self):
+        """
+        The standard deviation of this distribution.
+        """
+        return self.distribution.std(axis=-1)
+
+    @property
+    def pdf_var(self):
+        """
+        The variance of this distribution.
+        """
+        return self.distribution.var(axis=-1)
+
+    @property
+    def pdf_median(self):
+        """
+        The median of this distribution.
+        """
+        return np.median(self.distribution, axis=-1)
+
+    @property
+    def pdf_mad(self):
+        """
+        The median absolute deviation of this distribution.
+        """
+        return np.abs(self - self.pdf_median).pdf_median
+
+    # we set this by hand because the symbolic expression (below) requires scipy
+    # _smad_scale_factor = 1 / scipy.stats.norm.ppf(0.75)
+    _smad_scale_factor = 1.48260221850560203193936104071326553821563720703125
+
+    @property
+    def pdf_smad(self):
+        """
+        The median absolute deviation of this distribution rescaled to match the
+        standard deviation for a normal distribution.
+        """
+        return self.pdf_mad * self._smad_scale_factor
+
+    def percentiles(self, perc, **kwargs):
+        """
+        Compute percentiles of this Distribution.
+
+        Parameters
+        ----------
+        perc : float or array of floats
+            The desired  precentiles of the distribution (i.e., on [0,100]).
+        kwargs
+            Additional keywords are passed into `numpy.percentile`.
+
+        Returns
+        -------
+        percs : Quantity
+            The ``fracs`` percentiles of this distribution.
+        """
+        perc = np.percentile(self.distribution, perc, axis=-1)
+        # numpy.percentile strips units for unclear reasons, so we have to make
+        # a new object with units
+        if hasattr(self.distribution, '_new_view'):
+            return self.distribution._new_view(perc)
+        else:
+            return perc
+
+    def hist(self, maxtoshow=10, **kwargs):
```

Comment:
For this initial go, I'd remove this or at least make it a function rather than a method. To me it feels too arbitrary to be generally useful, and before you know it, there is a whole new API for setting graph properties attached to the class.